src/transformers/models/smolvlm/processing_smolvlm.py

```
@@ -161,17 +165,8 @@ def __init__(
         self.end_of_utterance_token = getattr(tokenizer, "end_of_utterance_token", "<end_of_utterance>")
         self.global_image_token = getattr(tokenizer, "global_image_token", "<global-img>")
         self.image_seq_len = image_seq_len
-
-        self.video_size = video_processor.video_sampling["video_size"]
-        self.image_size = image_processor.size
-
-        self.do_image_splitting = image_processor.do_image_splitting
-        self.do_video_splitting = video_processor.video_sampling.get("do_image_splitting", False)
-
-        self.default_max_frames = video_processor.video_sampling["max_frames"]
-        self.default_fps = video_processor.video_sampling["fps"]
-        # Matches one or more occurrences of <row_x_col_y> tags (where x and y are digits, optionally surrounded by newline characters
-        # self._regex_to_remove_extra_special_tokens = re.compile(r"(<row_\d+_col_\d+>\n?)+")
+        self.video_token = getattr(tokenizer, "video_token", "<video>")
+        chat_template = CHAT_TEMPLATE  # re-assign to the correct template for BC
```

Now, this is open to discussion. For BC in SmolVLMs we would need to replace input videos with `<video>` placeholders. Current chat templates on hub don't have it, because we used to manipulate conversation when applying chat template

This workaround is expected to work in most cases, unless users specifically save different chat template from official.Then it will cause confusion. 

Or should we move it to `apply_chat_template` whenever conversation has a "type=video" ?

---

IMO we should not override the chat template this way, as it could cause very surprising behaviour hard to understand/debug. So I think we have 2 solutions: 
- checking if the template is default (most cases) and if it's the case silently use this updated one, and raising a warning if it's not the default template explaining that it should handle video inputs correctly
- move it to `apply_chat_template` as you suggest

I would be in favor of the latter option as it seems more robust, but not sure how large a change it requires